src/main/java/co/rsk/federate/config/PowHSMBookkeepingConfig.java

```diff
@@ -1,87 +1,86 @@
-                this.difficultyCap = DIFFICULTY_CAP_MAINNET;
-                break;
-            case NetworkParameters.ID_TESTNET:
-                this.difficultyCap = DIFFICULTY_CAP_TESTNET;
-                break;
-            case NetworkParameters.ID_REGTEST:
-                this.difficultyCap = DIFFICULTY_CAP_REGTEST;
-                break;
-            default:
-                String message = "Invalid network specified for the Bookkeeping Config: " + networkParameter;
-                logger.error(message);
-                throw new IllegalArgumentException(message);
+        public BigInteger getDifficultyCap() {
+            return difficultyCap;
         }
     }
 
+    private static final BigInteger DEFAULT_DIFFICULTY_TARGET = BigInteger.valueOf(3);
+    private static final boolean DEFAULT_STOP_BOOKKEPING_SCHEDULER = false;
+    private static final int DEFAULT_MAX_AMOUNT_BLOCK_HEADERS = 7;
+    private static final int DEFAULT_MAX_CHUNK_SIZE = 10;
+    private static final long DEFAULT_INFORMER_INTERVAL = 2_000;
+
+    private final SignerConfig signerConfig;
+    private final String networkParameter;
+
+    public PowHSMBookkeepingConfig(SignerConfig signerConfig, String networkParameter) {
+        this.signerConfig = signerConfig;
+        this.networkParameter = networkParameter;
+    }
+
     public BigInteger getDifficultyTarget() {
-        return difficultyTarget;
+        return signerConfig.getConfig().hasPath("bookkeeping.difficultyTarget")
+            ? new BigInteger(signerConfig.getConfig().getString("bookkeeping.difficultyTarget"))
+            : DEFAULT_DIFFICULTY_TARGET;
     }
 
     public int getMaxAmountBlockHeaders() {
-        return maxAmountBlockHeaders;
+        return signerConfig.getConfig().hasPath("bookkeeping.maxAmountBlockHeaders")
+            ? signerConfig.getConfig().getInt("bookkeeping.maxAmountBlockHeaders")
+            : DEFAULT_MAX_AMOUNT_BLOCK_HEADERS;
     }
 
     public long getInformerInterval() {
-        return informerInterval;
+        return signerConfig.getConfig().hasPath("bookkeeping.informerInterval")
+            ? signerConfig.getConfig().getLong("bookkeeping.informerInterval")
+            : DEFAULT_INFORMER_INTERVAL;
     }
 
     public boolean isStopBookkeepingScheduler() {
-        return stopBookkeepingScheduler;
+        return signerConfig.getConfig().hasPath("bookkeeping.stopBookkeepingScheduler")
+            ? signerConfig.getConfig().getBoolean("bookkeeping.stopBookkeepingScheduler")
+            : DEFAULT_STOP_BOOKKEPING_SCHEDULER;
     }
 
     public int getMaxChunkSizeToHsm() {
-        return maxChunkSizeToHsm;
+        return signerConfig.getConfig().hasPath("bookkeeping.maxChunkSizeToHsm")
+            ? signerConfig.getConfig().getInt("bookkeeping.maxChunkSizeToHsm")
+            : DEFAULT_MAX_CHUNK_SIZE;
     }
 
     public BigInteger getDifficultyCap() {
-        return difficultyCap;
+        switch (networkParameter) {
```

Comment:
I believe the network can also be obtained from the device for HSM >= 4

Reply:
Nice, I will have this in mind for the next PR.